src/main/java/com/google/gfx/services/matching/MatchingService.java

```diff
@@ -0,0 +1,38 @@
+package com.google.gfx.services.matching;
+
+import com.google.gfx.model.Match;
+import com.google.gfx.model.Order;
+
+/**
+ * Matches existing orders with other existing or new ones.
+ */
+public interface MatchingService {
+
+  /**
+   * This method was conceived to be ran as part of a cron job that matches existing buy and sell
+   * orders.
+   */
+  void matchExistingBuyAndSellOrders();
+
+  /**
+   * Creates a match for an order, that must be accepted by the order owner.
+   *
+   * @param order original posted order
+   * @param amount amount of the match, which might fully or partially match the original order
+   */
+  void matchOrder(Order order, float amount);
+
+  /**
+   * Accepts a proposed match.
+   *
+   * @param match proposed match
+   */
+  void acceptMatch(Match match);
```

Comment:
I think you'd probably want some way for the caller to say whether the buyer or seller is accepting this Match, since both acceptances are tracked separately. 

More generally, it might be good to think about adding some sort of a Context object to all of these service methods, if for no other reason than to make it possible to do authorization. (i.e. user A shouldn't be able to accept an Match between users B and C).